roles/manage-ssh-private-keys/tasks/process-ssh-key.yml

```diff
@@ -0,0 +1,20 @@
+---
+
+- name: Create destination path {{ ssh_key.dest | dirname }}
+  file:
+    path: "{{ ssh_key.dest | dirname }}"
+    state: directory
+    owner: "{{ ssh_key.owner }}"
+    group: "{{ ssh_key.group }}"
+    mode: 0700
+  when:
+    - ssh_key.create_dest_dir is defined
+    - ssh_key.create_dest_dir
+
+- name: Copy SSH key {{ ssh_key.src | basename }}
```

Comment:
Maybe also add the dest directory here - i.e.: `Copy SSH key {{ ssh_key.src | basename }} to {{ ssh_key.dest }}`